runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -1435,3 +1427,33 @@ TR_JITaaSRelocationRuntime::allocateSpaceInDataCache(uintptr_t metaDataSize,
    return newDataStart;
    }
 
+uint8_t *
+TR_JITaaSRelocationRuntime::copyDataToCodeCache(const void *startAddress, size_t totalSize, TR_J9VMBase *fe)
+   {
+   TR::CompilationInfoPerThreadBase *compInfoPT = fe->_compInfoPT;
+   int32_t numReserved;
+   TR::CodeCache *codeCache = NULL;
+   TR::CodeCacheManager *manager = TR::CodeCacheManager::instance();
+   codeCache = manager->reserveCodeCache(false, totalSize, compInfoPT->getCompThreadId(), &numReserved);
+   if (!codeCache)
+      return NULL;
+
+   if (compInfoPT->getCompThreadId() >= 0 && fe->getCompilationShouldBeInterruptedFlag())
+      {
+      codeCache->unreserve();
+      return NULL;
```

Comment:
In this case we would do `failCompilation<TR::CodeCacheError>` which is not ideal because the real reason is compilation interrupted. I don't see an easy solution besides passing back an error code.

Reply:
I see that `TR_SharedCacheRelocationRuntime::allocateSpaceInCodeCache` used to pass back error code for that case, but it is commented out now. So mainline code has the same issue currently. It's probably fine then if we do the same.

Reply:
ok